src/structure.ts

```diff
@@ -53,97 +53,97 @@ export interface Components<Ctx extends Context = Context, Req extends Request =
-            ...acc,
-            [cmd.name]: cmd,
-        };
-    }, {}) as Commands<Ctx, Req, Sentry>;
+export const validateCommands = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmds: unknown[], warn = false) =>
+cmds.reduce<Commands<Ctx, Req, Sentry>>((acc, cmd) => {
+    if (!isCommand(cmd, warn)) return acc;
+
+    // Check the command doesn't already exist
+    if (acc[cmd.name]) {
+        if (warn)
+            console.warn(`Command ${cmd.name} already exists`);
+        return acc;
+    }
+
+    // Add the command
+    return {
+        ...acc,
+        [cmd.name]: cmd,
+    };
+}, {});
 
 /**
  * Validate that a given value is a {@link Component} object
  */
-const isComponent = (value: any, warn = false): value is Component => {
+function isComponent<Ctx extends Context, Req extends Request, Sentry extends Toucan | undefined>(value: unknown, warn: boolean): value is Component<Ctx, Req, Sentry> {
     if (typeof value !== 'object' || value === null) {
         if (warn)
             console.warn('Expected component to be an object');
         return false;
     }
 
-    if (typeof value.name !== 'string' || !value.name.length) {
+    if (!('name' in value) || typeof value.name !== 'string' || !value.name.length) {
         if (warn)
             console.warn('Expected component to have a name');
         return false;
     }
 
-    if (typeof value.execute !== 'function') {
+    if (!('execute' in value) || typeof value.execute !== 'function') {
         if (warn)
             console.warn('Expected component to have an execute function');
         return false;
     }
 
     return true;
-};
+}
 
 /**
  * Validate that a set of values are {@link Component} objects
  */
-export const validateComponents = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmps: any[], warn = false) =>
-    cmps.reduce((acc, cmp) => {
-        if (!isComponent(cmp, warn)) return acc;
-
-        // Check the component doesn't already exist
-        if (acc[cmp.name]) {
-            if (warn)
-                console.warn(`Component ${cmp.name} already exists`);
-            return acc;
-        }
-
-        // Add the component
-        return {
-            ...acc,
-            [cmp.name]: cmp,
-        };
-    }, {}) as Components<Ctx, Req, Sentry>;
+export const validateComponents = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmps: unknown[], warn = false) =>
+cmps.reduce<Components<Ctx, Req, Sentry>>((acc, cmp) => {
+    if (!isComponent(cmp, warn)) return acc;
+
+    // Check the component doesn't already exist
+    if (acc[cmp.name]) {
+        if (warn)
+            console.warn(`Component ${cmp.name} already exists`);
+        return acc;
+    }
+
+    // Add the component
+    return {
+        ...acc,
+        [cmp.name]: cmp,
+    };
+}, {});
```

Comment:
```suggestion
    cmps.reduce<Components<Ctx, Req, Sentry>>((acc, cmp) => {
        if (!isComponent(cmp, warn)) return acc;

        // Check the component doesn't already exist
        if (acc[cmp.name]) {
            if (warn)
                console.warn(`Component ${cmp.name} already exists`);
            return acc;
        }

        // Add the component
        return {
            ...acc,
            [cmp.name]: cmp,
        };
    }, {});
```
Looks like some indentation was lost here